spy/backend/spy.py

```diff
@@ -367,6 +367,21 @@ def fmt_expr_CmpOp(self, op: ast.CmpOp) -> str:
             r = f"({r})"
         return f"{l} {op.op} {r}"
 
+    def fmt_expr_CmpChain(self, chain: ast.CmpChain) -> str:
+        assert chain.comparisons
+        pieces: list[str] = []
+        first = chain.comparisons[0]
+        left_str = self.fmt_expr(first.left)
+        if first.left.precedence < chain.precedence:
+            left_str = f"({left_str})"
+        pieces.append(left_str)
+        for cmp in chain.comparisons:
+            right_str = self.fmt_expr(cmp.right)
+            if cmp.right.precedence < chain.precedence:
+                right_str = f"({right_str})"
+            pieces.append(f"{cmp.op} {right_str}")
+        return " ".join(pieces)
+
```

Comment:
this should have a dedicated test in `test_backend_spy`.
The error which you most likely saw during the development is just a sanity check, which ensures that the spy backend supports all the ast nodes which are seen while running the tests.
However, the sanity check doesn't test that the formatting is actually correct, that' why we need a dedicated test.